ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Reader.hs

```diff
@@ -415,14 +415,15 @@ forward ::
   -> [Point blk]
   -> m (Maybe (Point blk))
 forward registry varReader blockComponent CDB{..} = \pts -> do
-    -- NOTE: we don't use 'Query.getCurrentChain', which only returns the last
-    -- @k@ headers, because we want to see the headers that have not yet been
-    -- written to the ImmutableDB too.
-    (curChain, readerState) <- atomically $
-      (,) <$> readTVar cdbChain <*> readTVar varReader
-    slotNoAtImmutableDBTip <- atomically $
-      ImmutableDB.getTipSlot cdbImmutableDB
-    findFirstPointOnChain curChain readerState slotNoAtImmutableDBTip pts
+    -- NOTE: we use `cdbChain` instead of 'Query.getCurrentChain', which only
+    -- returns the last @k@ headers, because we need to also see the headers
+    -- that happen to have not yet been GC'd over to the ImmutableDB.
```

Comment:
```suggestion
    -- that happen to have not yet been copied over to the ImmutableDB.
```
They are first copied and after a delay of 60s GC'd. It's the copying that is important here, not the GC'ing.